File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int sum = 0;
        int cnt = 0;

        Map<Integer, Integer> map = new HashMap<>();

        map.put(0, 1);

        for(int num : nums){
            sum += num;
        int rem = sum - k;

        cnt += map.getOrDefault(rem, 0);

        map.put(sum, map.getOrDefault(sum, 0)+1);
        }
        return cnt;
    }
    
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        freq = {0: 1}
        curr = 0
        cnt = 0

        for num in nums:
            curr += num

            if (curr-k) in freq:
                cnt += freq[curr-k]
            freq[curr] = freq.get(curr, 0)+1

        return cnt